examples/mnist_net2net.py

```diff
@@ -0,0 +1,381 @@
+    assert new_width > teacher_w1.shape[1], (
+        "new width (nout) should be bigger than the existing one")
+
+    n = new_width - teacher_w1.shape[1]
+    if init == 'random-pad':
+        new_w1 = np.random.normal(0, 0.1, size=(teacher_w1.shape[0], n))
+        new_b1 = np.ones(n) * 0.1
+        new_w2 = np.random.normal(0, 0.1, size=(n, teacher_w2.shape[1]))
+    elif init == 'net2wider':
+        index = np.random.randint(teacher_w1.shape[1], size=n)
+        factors = np.bincount(index)[index] + 1.
+        new_w1 = teacher_w1[:, index]
+        new_b1 = teacher_b1[index]
+        new_w2 = teacher_w2[index, :] / factors[:, np.newaxis]
+    else:
+        raise ValueError("Unsupported weight initializer: %s" % init)
+
+    student_w1 = np.concatenate((teacher_w1, new_w1), axis=1)
+    if init == 'random-pad':
+        student_w2 = np.concatenate((teacher_w2, new_w2), axis=0)
+    elif init == 'net2wider':
+        # add small noise to break symmetry, so that student model will have
+        # full capacity later
+        noise = np.random.normal(0, 5e-2 * new_w2.std(), size=new_w2.shape)
+        student_w2 = np.concatenate((teacher_w2, new_w2 + noise), axis=0)
+        student_w2[index, :] = new_w2
+    student_b1 = np.concatenate((teacher_b1, new_b1), axis=0)
+
+    return student_w1, student_b1, student_w2
+
+
+def deeper2net_conv2d(teacher_w):
+    """Get initial weights for a deeper conv2d layer by net2deeper'.
+
+    # Auguments
+        teacher_w: `weight` of previous conv2d layer,
+          of shape (nb_filter, nb_channel, kh, kw)
+    """
+    nb_filter, nb_channel, kh, kw = teacher_w.shape
+    student_w = np.zeros((nb_filter, nb_filter, kh, kw))
+    for i in xrange(nb_filter):
+        student_w[i, i, (kh - 1) / 2, (kw - 1) / 2] = 1.
+    student_b = np.zeros(nb_filter)
+    return student_w, student_b
+
+
+def copy_weights(teacher_model, student_model, layer_names):
+    """Copy weights from teacher_model to student_model,
+     for layers with names listed in layer_names
+    """
+    for name in layer_names:
+        weights = teacher_model.get_layer(name=name).get_weights()
+        student_model.get_layer(name=name).set_weights(weights)
+
+# methods to construct teacher_model and student_models
+
+
+def make_teacher_model(train_data, validation_data, nb_epoch=3):
+    """Train a simple CNN as teacher model.
+    """
+    model = Sequential()
+    model.add(Conv2D(64, 3, 3, input_shape=input_shape,
+                     border_mode="same", name="conv1"))
+    model.add(MaxPooling2D(name="pool1"))
+    model.add(Conv2D(128, 3, 3, border_mode="same", name="conv2"))
+    model.add(MaxPooling2D(name="pool2"))
+    model.add(Flatten(name="flatten"))
+    model.add(Dense(128, activation="relu", name="fc1"))
+    model.add(Dense(nb_class, activation="softmax", name="fc2"))
+    model.compile(loss="categorical_crossentropy",
+                  optimizer="sgd", metrics=["accuracy"])
```

Comment:
You shouldn't use SGD with default parameters. Prefer RMSprop or SGD with momentum and a tuned LR.